src/rules/netlify-role-management.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * @title Netlify Role Management
+ * @overview Adds a default role if the user doesn't have any yet and attaches roles to the ID Token.
+ * @gallery true
+ * @category marketplace
+ *
+ *
+ * **Required configuration** (this Rule will be skipped if any of the below are not defined):
+ *
+ * **Optional configuration:**
+ *
+ *    - `DEFAULT_ROLE_NAME` - name of the default role to be given to a user
+ *    - `DEFAULT_ROLE_ID` - id of the role to be given to a user
+ *    - `CUSTOM_CLAIMS_NAMESPACE` - namespace for adding custom claims to ID Token
+ */
+
+async function (user, context, callback) {
+
+  const ManagementClient = require('auth0@2.17.0').ManagementClient;
+  
+  const namespace = configuration.CUSTOM_CLAIMS_NAMESPACE || "https://netlify-integration.com";
+  const assignedRoles = (context.authorization || {}).roles;
+  const defaultRoleName = configuration.DEFAULT_ROLE_NAME;
+  const defaultRoleId = configuration.DEFAULT_ROLE_ID;
+  
+  //give default role if the user doesn't already have any roles assigned
+  if( (!assignedRoles || assignedRoles.length === 0) && defaultRoleName && defaultRoleId) {
```

Comment:
Do we also want to add the default role if someone doesn't have it? So if someone is only "admin" we also make sure they have the default?

Reply:
Ahh interesting. I imagine we could skip this though. If the user doesn't have a role of "guest" that means that were manually added in Auth0. At that point, I would expect the admin did it for a specific reason. Otherwise, the user would have gotten the guest role originally.

Reply:
Also, if we specifically didn't want someone to have the "guest" role, we wouldn't have any way to prevent them from having it.